contrib/mesos/pkg/executor/broadcast.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package executor
+
+import (
+	"errors"
+	"math/big"
+	"math/rand"
+	"sync"
+)
+
+type Sink interface {
+	Write(*RegisteredPod)
+}
+
+type SinkFunc func(*RegisteredPod)
+
+func (sf SinkFunc) Write(x *RegisteredPod) {
+	sf(x)
+}
+
+type broadcast struct {
+	in        <-chan *RegisteredPod
+	out       map[int]Sink
+	lock      sync.Mutex
```

Comment:
what does the lock protect? The whole object or just the slots variable? Better style:
```
a
b

lock
c
d
```

where lock protects c and d only.